core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -715,7 +714,9 @@ private[spark] object SparkConf extends Logging {
     PRINCIPAL.key -> Seq(
       AlternateConfig("spark.yarn.principal", "3.0")),
     KERBEROS_RELOGIN_PERIOD.key -> Seq(
-      AlternateConfig("spark.yarn.kerberos.relogin.period", "3.0"))
+      AlternateConfig("spark.yarn.kerberos.relogin.period", "3.0")),
+    KERBEROS_FILESYSTEMS_TO_ACCESS.key -> Seq(
+      AlternateConfig(YARN_FILESYSTEMS_TO_ACCESS.key, "3.0"))
```

Comment:
Instead of keeping `YARN_FILESYSTEMS_TO_ACCESS`, just have the key here and delete the constant. Also add the old `"spark.yarn.access.namenodes"` as an alternative (keeping the version where it was deprecated).

Reply:
Fixed. Just to learn how to decide next time why is it different scenario than `spark.yarn.access.namenodes` and `spark.yarn.access.hadoopFileSystems`? I wanted to make a shift here (oldest dropped, others moving to the original 2 slots).

Reply:
I don't think we have a policy for dropping configs yet, so better to keep things working unless there's a good reason not to.